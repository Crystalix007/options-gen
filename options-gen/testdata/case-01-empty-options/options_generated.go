// Code generated by options-gen. DO NOT EDIT.
package testcase

import (
	uniqprefixformultierror "github.com/hashicorp/go-multierror"
)

type optOptionsMeta struct {
	setter    func(o *Options)
	validator func(o *Options) error
}

func NewOptions(

	options ...optOptionsMeta,
) Options {
	o := Options{}

	for i := range options {
		options[i].setter(&o)
	}

	return o
}

func (o *Options) Validate() error {
	var g uniqprefixformultierror.Group

	return g.Wait().ErrorOrNil()
}
