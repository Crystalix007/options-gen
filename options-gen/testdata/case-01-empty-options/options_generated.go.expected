// Code generated by options-gen. DO NOT EDIT.
package testcase

type OptOptionsSetter func(o *Options)

func NewOptions(
	options ...OptOptionsSetter,
) Options {
	o := Options{}

	for _, opt := range options {
		opt(&o)
	}
	return o
}

func (o *Options) Validate() error {
	return nil
}
